File: evals/behavioral/tasks/rust/fixtures/typestate-builder.bad.rs
// Runtime-checked builder: one type holds Option fields and `build` is always
// callable, deferring the "url required" check to a runtime panic. `new().build()`
// type-checks (and would panic), so the invariant is NOT enforced by the types.
pub struct RequestBuilder {
    url: Option<String>,
    method: Option<String>,
}

impl RequestBuilder {
    pub fn new() -> RequestBuilder {
        RequestBuilder { url: None, method: None }
    }
    pub fn url(mut self, url: String) -> RequestBuilder {
        self.url = Some(url);
        self
    }
    pub fn method(mut self, method: String) -> Self {
        self.method = Some(method);
        self
    }
    pub fn build(self) -> Request {
        Request { url: self.url.expect("url is required"), method: self.method }
    }
}

pub struct Request {
    pub url: String,
    pub method: Option<String>,
}

File: evals/behavioral/tasks/scala3/fixtures/typestate-builder.bad.scala
// Runtime-checked builder: one type holds Option fields and `build` is always
// callable, deferring the "url required" check to a runtime throw.
// `RequestBuilder().build()` compiles (and would throw), so the invariant is
// NOT enforced by the types.
final case class Request(url: String, method: Option[String])

final class RequestBuilder {
  private var urlOpt: Option[String] = None
  private var methodOpt: Option[String] = None

  def url(url: String): RequestBuilder = { urlOpt = Some(url); this }
  def method(method: String): RequestBuilder = { methodOpt = Some(method); this }
  def build(): Request =
    Request(urlOpt.getOrElse(throw new IllegalStateException("url is required")), methodOpt)
}

object RequestBuilder {
  def apply(): RequestBuilder = new RequestBuilder
}
